src/Client/DTO/WorkflowExecutionInfo.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Temporal\Client\DTO;
+
+use DateTimeInterface;
+use JetBrains\PhpStorm\Immutable;
+use Temporal\Client\DTO\ResetPointInfo as ResetPointInfoDto;
+use Temporal\DataConverter\EncodedCollection;
+use Temporal\DataConverter\ValuesInterface;
+use Temporal\Workflow\WorkflowExecution;
+use Temporal\Workflow\WorkflowType;
+
+/**
+ * @psalm-immutable
+ */
+#[Immutable]
+final class WorkflowExecutionInfo
+{
+    /**
+     * @param array<ResetPointInfoDto> $autoResetPoints
+     */
+    public function __construct(
+        public WorkflowExecution $execution,
+        public WorkflowType $type,
+        public ?DateTimeInterface $startTime,
+        public ?DateTimeInterface $closeTime,
+        public int $status,
```

Comment:
The status here should be some kind of well-defined enum based on https://github.com/temporalio/api/blob/5f21c72de0ec9e6037e84c95f097641b71d7c39f/temporal/api/enums/v1/workflow.proto#L74

Reply:
PHP Enums are available since PHP 8.1.
PHP 8.1 is required by interceptors.
We can bump PHP version right now to ^8.1, or use constants and it will look like deprecation after the interceptors release

Reply:
Ah, I see. I don't have a sense of how disruptive the PHP version bump would be. If that's unlikely to be a big problem for anyone I'd say let's go for that, otherwise I'm fine leaving it a scalar (with predefined constants) for now.

Reply:
Any way we have to up to 8.1 with the interceptors release